### ISSUE TITLE
New chat button covers the red delete chat button  

### ISSUE THREAD

**Issue by lukaszfryc** [status-im/status-react#5043]
### Description

*Type*: Bug

*Summary*: Floating "+" button covers the delete button:

<img src="https://user-images.githubusercontent.com/7532782/42265385-aebf549c-7f73-11e8-94ff-8581e279f20a.png" width="200">

### Reproduction

- Open Status
- Have long list of chats
- Swipe a chat on the bottom to delete it 

### Additional Information
* Status version: nightly, 4 July
* Operating System: Android, iOS

### UX insight
In ux research (RRR8) we found that longpress wroked for reply/copy message with all participants. In the few cases it needed multiple attempts it showed to be easily discoverable. 
https://docs.google.com/presentation/d/10qb5jRXwRT9lYkxrn8Ui5Bov_JRuf1WBaPqSXd0ZAtY/edit#slide=id.g3f411e60e9_0_155

**Comment by lukaszfryc** [comment]
@errorists any idea how this could be improved?

**Comment by errorists** [comment]
What we could do is create an exception that if you swipe a cell that is partially covered by a FAB, the FAB would move its position 1:1 along with the cell (animate it on a X axis). But a better alternative would be to change the swipe mechanism so that instead of tapping the uncovered button to delete, you'd only have to swipe for a set minimum distance: after you'd swiped for say 60 points we'd register the delete action. It's popular among apps I use and allows for faster triage of unwanted chats/dapps, if hard to explain I can create a prototype for this behaviour. What you think @andmironov ?

**Comment by errorists** [comment]
I've illustrated my proposed solution with a prototype, I've also added a long-press menu if redesigning the swipe behaviour seems like an overkill https://framer.cloud/CEGLr

<blockquote><img src="https://framer.cloud/CEGLr/2/framer/social-1200x630.png" width="48" align="right"><div><strong><a href="https://framer.cloud/CEGLr">Chat List</a></strong></div><div>This project was made with Framer, a design tool for turning original concepts into working prototypes.</div></blockquote>

**Comment by lukaszfryc** [comment]
@errorists I like the long-press interaction more than swipe to delete. Here is my reasoning:

1. In the near future, we may want to consider swiping between Home, Wallet and Profile tabs on Android. If so, swipe to delete may interfere with swiping between tabs
2. Long-press can display menu with additional options like "Mark as read" or "Pin to the top". It's a good place to display them. Telegram does it and it does not have swipe gesture.
3. We've seen some bug reports about our swipe to delete gesture. E.g. that it brakes UI when swiped to the left and then back to the right
4. If we go with swipe to delete, I'd ask for additional confirmation (show a modal). Alternatively, show a status bar with a revert button

cc @andmironov

**Comment by errorists** [comment]
@lukaszfryc All good points! 

1. I've considered this heavily, but in the context of having separate tabs inside the Home screen, for example swiping between Chats, Group Chats and Dapps.
2. Technically we could allow for additional actions inside the swipe gesture, exposing different buttons by swiping from left and right. Depending on how often you access those actions, gestures might be more convenient, but that's not saying either is worse, just more appropriate for different actions.
3. Agree there, our current implementation is lacking. It doesn't follow the finger properly, it's jerky in movement and buggy, when dragged past its bounds a white rectangle would slowly cover the text. That would be my main reason to go with a menu, unless we can make the swipe gesture nicely designed and implemented we shouldn't be introducing them at all.
4. Agree, I've already updated it with an alert prompt after feedback but a snackbar with Undo button would be even nicer.

**Comment by lukaszfryc** [comment]
> 1. I've considered this heavily, but in the context of having separate tabs inside the Home screen, for example swiping between Chats, Group Chats and Dapps.

Good idea. It's how Facebook Messenger works. And, because of that, the app uses long press instead of swipe to delete a chat. 

> 2. Technically we could allow for additional actions inside the swipe gesture, exposing different buttons by swiping from left and right. Depending on how often you access those actions, gestures might be more convenient, but that's not saying either is worse, just more appropriate for different actions.

Swipe gesture maybe a bit more convenient if you have lot of items to remove. But only a little since you need to confirm each action anyway (delete dialog). If you want to remove many chats or dapps quickly, I'd consider some bulk action (option to select many items). But, it's not needed now.
 
> 4. Agree, I've already updated it with an alert prompt after feedback but a snackbar with Undo button would be even nicer.

I'd keep it simple now and use dialog. We already show it when you want to remove a chat while being on the chat screen. You click chat menu, press delete and confirm in the dialog.

----

@errorists summing up, do you think we should remove swipe to delete and go with long press? Can we have opinion of other designers, @andmironov?

**Comment by errorists** [comment]
Summing up, I'm in favour of removing the swipe gesture in its current implementation and replacing it with a long press action.

**Comment by denis-sharypin** [comment]
Shame on me but I have just discovered that you can remove chats with the long press on FB messenger. Can it be considerated as a not popular behavior?  I like the idea of adding some new options to the context menu though. 

I personally would try to focus on current problems and needs. Yeah, maybe in the near future we will need a swipe between home tab categories, but most likely we will create a separate tab for dapps and browser. Nobody sure yet.  

To summarize: I would use this to solve our particular problem here.  

> Only have to swipe for a set minimum distance: after you'd swiped for say 60 points we'd register the delete action. It's popular among apps I use and allows for faster triage of unwanted chats/dapps

**Comment by lukaszfryc** [comment]
> Shame on me but I have just discovered that you can remove chats with the long press on FB messenger. Can it be considerated as a not popular behavior?

I don't think so. Messenger and Telegram use it.

**Comment by denis-sharypin** [comment]
My telegram client doesn’t have it :(

**Comment by jpbowen** [comment]
To try and help move this conversation along, what are some of the actions we expect people to be able to do on long press for a chat (and we have to consider DApps)? Some things that come to mind: 
- Delete
- Mark as unread
- Pin (particularly helpful for group chat?) 
- Ignore / Block
- Mute / turn notifications off
- ...?

All of which are useful but have not heard that any of these things are particularly necessary or painful atm other than 'Delete' which has been covered in this issue. Both are learned behaviors as is and, unfortunately, splintered across different platforms. Could we implement the swipe gesture as @errorists has outlined above - i.e. 60 points + final confirmation - with a principle that we use it for destructive actions (more context: https://www.nngroup.com/articles/contextual-swipe/)? And a plan to follow up with long press in the future for some of these key actions. Wdyt? 

<blockquote><img src="https://media.nngroup.com/media/articles/opengraph_images/Slide057.png" width="48" align="right"><div>Nielsen Norman Group</div><div><strong><a href="https://www.nngroup.com/articles/contextual-swipe/">Using Swipe to Trigger Contextual Actions</a></strong></div><div>Implementations of swipe-to-delete or swipe to reveal contextual actions often suffer from usability problems.</div></blockquote>

**Comment by hesterbruikman** [comment]
In ux research (RRR8) we found that longpress wroked for reply/copy message with all participants. In the few cases it needed multiple attempts it showed to be easily discoverable. https://docs.google.com/presentation/d/10qb5jRXwRT9lYkxrn8Ui5Bov_JRuf1WBaPqSXd0ZAtY/edit#slide=id.g3f411e60e9_0_155

<blockquote><img src="https://lh6.googleusercontent.com/_vO60E7Z21r_Wt8D2EvcpUodrA1wl1klLexYshNEYeZK_zD30zGKxUXmt4JEUnlGwjOVjw=w1200-h630-p" width="48" align="right"><div>Google Docs</div><div><strong><a href="https://docs.google.com/presentation/d/10qb5jRXwRT9lYkxrn8Ui5Bov_JRuf1WBaPqSXd0ZAtY/edit?usp=embed_facebook">Report RRR8 Chat & Command</a></strong></div><div>Hester Status 17 / 09 / 2018 UXR findings Chat & Command RRR8 Sept 6th 2018</div></blockquote>
